pkg/agent/attestor/node/experimental_test.go

```diff
@@ -0,0 +1,445 @@
+			bootstrapBundle: caCert,
+			svid:            svid,
+			bundle:          bundle,
+			err:             "error in CloseSend",
+			closeSendErr:    errors.New("error in CloseSend"),
+		},
+		{
+			name:            "get bundle error",
+			bootstrapBundle: caCert,
+			svid:            svid,
+			err:             "error in GetBundle",
+			getBundleErr:    errors.New("error in GetBundle"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			require := require.New(t)
+
+			// prepare the temp directory holding the cached bundle/svid
+			svidCachePath, bundleCachePath, removeDir := prepareTestDir(t, testCase.cachedSVID, testCase.cachedBundle)
+			defer removeDir()
+
+			// load up the fake agent-side node attestor
+			agentNA, agentNADone := prepareAgentNA(t, fakeagentnodeattestor.Config{
+				Fail:      testCase.failFetchingAttestationData,
+				Responses: testCase.challengeResponses,
+			})
+			defer agentNADone()
+
+			// load up the fake server-side node attestor
+			serverNA, serverNADone := prepareServerNA(t, fakeservernodeattestor.Config{
+				TrustDomain: "domain.test",
+				Data: map[string]string{
+					"TEST": "foo",
+				},
+				Challenges: map[string][]string{
+					"foo": testCase.challengeResponses,
+				},
+			})
+			defer serverNADone()
+
+			// load up an in-memory key manager
+			km, kmDone := prepareKeyManager(t, testCase.storeKey)
+			defer kmDone()
+
+			// initialize the catalog
+			catalog := fakeagentcatalog.New()
+			catalog.SetNodeAttestor(fakeagentcatalog.NodeAttestor("test", agentNA))
+			catalog.SetKeyManager(fakeagentcatalog.KeyManager(km))
+
+			// kick off the gRPC server serving the node API
+			serverAddr, serverDone := startNodeServer(t, tlsConfig, fakeNodeAPIConfig{
+				CACert:         caCert,
+				Attestor:       serverNA,
+				FailAttestCall: testCase.failAttestCall,
+			})
+			defer serverDone()
+
+			// create the attestor
+			log, _ := test.NewNullLogger()
+			ta := &testAttestor{
+				agentClient:  &fakeAgentClient{},
+				bundleClient: &fakeBundleClient{},
+			}
+			attestor := newAttestor(&Config{
+				Catalog:         catalog,
+				Metrics:         telemetry.Blackhole{},
+				JoinToken:       testCase.joinToken,
+				SVIDCachePath:   svidCachePath,
+				BundleCachePath: bundleCachePath,
+				Log:             log,
+				TrustDomain: url.URL{
+					Scheme: "spiffe",
+					Host:   "domain.test",
+				},
+				TrustBundle:            makeTrustBundle(testCase.bootstrapBundle),
+				InsecureBootstrap:      testCase.insecureBootstrap,
+				ServerAddress:          serverAddr,
+				ExperimentalAPIEnabled: true,
+			})
+			attestor.createNewAgentClient = func(conn *grpc.ClientConn) agentpb.AgentClient {
+				return ta.agentClient
+			}
+			attestor.createNewBundleClient = func(conn *grpc.ClientConn) bundlepb.BundleClient {
+				return ta.bundleClient
+			}
+
+			ta.agentClient.joinToken = testCase.joinToken
```

Comment:
maybe you can add agentClient to tt instead all separated value?and just set it?

Reply:
same bundle